scvi/inference/posterior.py

```diff
@@ -968,20 +984,19 @@ def within_cluster_degenes(
             cell_labels = self.gene_dataset.labels.ravel()
         de_res = []
         de_cluster = []
-        states = np.asarray([1 if x else 0 for x in states])
-        nstates = np.asarray([0 if x else 1 for x in states])
+        oppo_states = ~states
```

Comment:
can you comment the docstring or add one quick comment to explain what those things are doing?

Reply:
It looks like just refactoring but I'm not sure

Reply:
this is the same code that's in master, it fixes a bug where only cell 0 and cell 1 were being compared because they were ints.

Reply:
I see OK